### PR TITLE
Budget app - Capstone Project

### PR DESCRIPTION
**Pull request by orozCoding**
I created a Ruby on Rails application that allows the user to:

- register and log in, so that the data is private to them.
- introduce new transactions associated with a category.
- see the money spent on each category.

Built with Ruby on Rails and with required gems such as Devise and CanCanCan.

The app has proper authentication and authorization features.
Wrote proper unit and integration tests for each model, controller, and view.
Implemented all the pages and features required.
Followed the design template provided

Loom video:
https://www.loom.com/share/7af7cbe57e0c4b9aaaa34b8b11ce7bd3

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    end
+
+    it 'is invalid if names are empty' do
+      @user.first_name = nil
+      expect(@user).to_not be_valid
+      @user.first_name = 'Amigo'
+      @user.last_name = nil
+      expect(@user).to_not be_valid
+    end
+
```
Comment (uwadonat) [inline spec/models/user_spec.rb:18]: - [ ] Some of your tests are failing. Please fix this.
Screenshot from 2022-05-24 15-21-18

**Review by uwadonat — CHANGES_REQUESTED** [review]
Hi **[orozCoding](https://github.com/orozCoding),** :hand: 

Good job so far! :mechanical_arm: 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>
### To highlight! :+1: 
- Your commit and pull request messages are very descriptive ✔️
- Your code is well organized and clean  ✔️
- You did extra mile by adding edit option ✔️
almost
 
## Status:Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/models/user_spec.rb (hunk 1 above)
Comment (orozCoding) [reply spec/models/user_spec.rb:18]: Thank you! @uwadonat 

For some reason, the tests failed just on the first try. After repeating the `rspec` command, everything worked.

I edited so it runs correctly from the first try.

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @orozCoding 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Great job working on the required changes.
- Nice communication with the previous reviewer.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**